app/controllers/friend_requests_controller.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+class FriendRequestsController < ApplicationController
+  before_action :set_friend_request, only: %i[update destroy]
+
+  def index
+    @sent_requests = current_user.friend_requests_sent.pending
+    @received_requests = current_user.friend_requests_received.pending
+  end
+
+  def create
+    @friend_request = FriendRequest.find_or_initialize_by(sender_id: current_user.id,
+                                                          receiver_id: params[:friend_request][:receiver_id])
+    @friend_request.status = FriendRequest.statuses[:pending]
+    respond_to do |format|
+      if @friend_request.save
+        format.html do
+          redirect_back fallback_location: root_path,
```

Comment:
To remove duplication here perhaps we can extract `redirect_back fallback_location: root_path` outside of the conditionals. Anyway whether `@friend_request.save` is successful or not we are redirecting to root path.